run_unittests.py

```diff
@@ -2978,47 +2978,61 @@ def test_guessed_linker_dependencies(self):
-            # .lib as extension. For a DLL the import library is installed as .lib. Thus for msvc
-            # this tests needs to use shared libraries to test the path resolving logic in the
-            # dependency generation code path.
-            extra_args = ['--default-library', 'static']
-        self.init(testdirlib, extra_args=extra_args)
-        self.build()
-        self.install()
-        libbuilddir = self.builddir
-        installdir = self.installdir
-        libdir = os.path.join(self.installdir, self.prefix.lstrip('/').lstrip('\\'), 'lib')
 
-        # build user of library
-        self.new_builddir()
-        # replace is needed because meson mangles platform pathes passed via LDFLAGS
-        self.init(os.path.join(testdirbase, 'exe'),
-                  override_envvars={"LDFLAGS": '-L{}'.format(libdir.replace('\\', '/'))})
-        self.build()
-        self.assertBuildIsNoop()
+        libdir_flags = ['-L']
+        if is_windows():
+            libdir_flags += ['/LIBPATH:']
 
-        # rebuild library
-        exebuilddir = self.builddir
-        self.installdir = installdir
-        self.builddir = libbuilddir
-        # Microsoft's compiler is quite smart about touching import libs on changes,
-        # so ensure that there is actually a change in symbols.
-        self.setconf('-Dmore_exports=true')
-        self.build()
-        self.install()
-        # no ensure_backend_detects_changes needed because self.setconf did that already
+        initial_builddir = self.builddir
+        initial_installdir = self.installdir
+
+        for libdir_flag in libdir_flags:
+            # build library
+            extra_args = None
+            env = get_fake_env(testdirlib, self.builddir, self.prefix)
+            if env.detect_c_compiler(MachineChoice.HOST).get_id() not in {'msvc', 'clang-cl', 'intel-cl'}:
+                # static libraries are not linkable with -l with msvc because meson installs them
+                # as .a files which unix_args_to_native will not know as it expects libraries to use
+                # .lib as extension. For a DLL the import library is installed as .lib. Thus for msvc
+                # this tests needs to use shared libraries to test the path resolving logic in the
+                # dependency generation code path.
+                extra_args = ['--default-library', 'static']
+            self.init(testdirlib, extra_args=extra_args)
+            self.build()
+            self.install()
+            libbuilddir = self.builddir
+            installdir = self.installdir
+            libdir = os.path.join(self.installdir, self.prefix.lstrip('/').lstrip('\\'), 'lib')
+
+            # build user of library
+            self.new_builddir()
+            # replace is needed because meson mangles platform pathes passed via LDFLAGS
+            self.init(os.path.join(testdirbase, 'exe'),
+                    override_envvars={"LDFLAGS": '{}{}'.format(libdir_flag,libdir.replace('\\', '/'))})
```

Comment:
__[Flake8]__ [E128] continuation line under-indented for visual indent ([link](https://sider.review/gh/repos/19784232/pulls/5881#issue-4509459))

<sub>posted by Sider</sub>